crates/vault-registry/src/migration.rs

```diff
@@ -0,0 +1,320 @@
+                weight.saturating_accrue(Runtime::DbWeight::get().reads_writes(12, 7));
+            }
+
+            StorageVersion::new(1).put::<Pallet<Runtime>>();
+            weight.saturating_add(Runtime::DbWeight::get().reads_writes(1, 2))
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade(state: Vec<u8>) -> Result<(), &'static str> {
+            let rewards_state: RewardsState<SignedFixedPoint<Runtime>> =
+                Decode::decode(&mut state.as_slice()).expect("invalid state generated by pre_upgrade");
+
+            let stake_entries_after: u32 = reward::Stake::<Runtime, VaultRewardsInstance>::iter().count() as u32;
+            log::info!(
+                target: TARGET,
+                "number of stake entries after: {:?}",
+                stake_entries_after
+            );
+            ensure!(
+                stake_entries_after == rewards_state.stake_entries,
+                "Not all entries were migrated"
+            );
+
+            ensure!(
+                reward::TotalRewards::<Runtime, VaultRewardsInstance>::get(
+                    <Runtime as currency::Config>::GetNativeCurrencyId::get()
+                ) == Zero::zero(),
+                "All rewards should be zero"
+            );
+            ensure!(
+                reward::TotalRewards::<Runtime, VaultRewardsInstance>::get(
+                    <Runtime as currency::Config>::GetWrappedCurrencyId::get()
+                ) == Zero::zero(),
+                "All rewards should be zero"
+            );
+
+            let native_staking_rewards_after = staking::TotalRewards::<Runtime>::iter_prefix_values(
+                <Runtime as currency::Config>::GetNativeCurrencyId::get(),
+            )
+            .reduce(|a, b| a.saturating_add(b))
+            .unwrap_or_default();
+            log::info!(
+                target: TARGET,
+                "rewards (native) before: {:?}, after: {:?}",
+                rewards_state.total_rewards_native,
+                native_staking_rewards_after,
+            );
+            ensure!(
+                native_staking_rewards_after == rewards_state.total_rewards_native,
+                "Previous rewards should be in staking"
+            );
+
+            let wrapped_staking_rewards_after = staking::TotalRewards::<Runtime>::iter_prefix_values(
+                <Runtime as currency::Config>::GetWrappedCurrencyId::get(),
+            )
+            .reduce(|a, b| a.saturating_add(b))
+            .unwrap_or_default();
+            log::info!(
+                target: TARGET,
+                "rewards (wrapped) before: {:?}, after: {:?}",
+                rewards_state.total_rewards_wrapped,
+                wrapped_staking_rewards_after,
+            );
+            ensure!(
+                wrapped_staking_rewards_after == rewards_state.total_rewards_wrapped,
+                "Previous rewards should be in staking"
+            );
+
+            Ok(())
```

Comment:
I'd like to have some general integrity checks, that check that the `capacity`, `reward`, `staking` and `TotalUserVaultCollateral` entries are consistent. This will be useful to check the correctness of the migration, but will also be useful to continuously check the integrity using the new [try-state](https://github.com/paritytech/substrate/blob/ee3c159829398add73a7c26fd7cba6de1c51257b/frame/support/src/traits/try_runtime.rs#L90)